src/syncPlugin.ts

```diff
@@ -0,0 +1,124 @@
+import { Plugin, PluginKey, Selection } from "prosemirror-state"
+import { next as am } from "@automerge/automerge"
+import { DocHandle, DocHandleChangePayload } from "@automerge/automerge-repo"
+import pmToAm from "./pmToAm"
+import amToPm from "./amToPm"
+import { docFromSpans } from "./traversal"
+import { patchesToTr } from "./patchesToTr"
+import { ChangeSet } from "prosemirror-changeset"
+import { SchemaAdapter } from "./schema"
+import { isArrayEqual } from "./utils"
+
+export const syncPluginKey = new PluginKey("automerge-sync")
```

Comment:
Is this a sufficiently unique name? I can see people using this name as a plugin key whilst testing things and then getting very confused.

Reply:
The name for a plugin key doesn't have to be unique; [ProseMirror will make one for you](https://github.com/ProseMirror/prosemirror-state/blob/09ffa0c8b5a331f6fbec2809b5bd454e4a43744d/src/plugin.ts#L134). With plugin keys, the only way to access them and their plugin state would then be by importing the plugin key itself, and using it directly.

For example, `tr.getMeta(syncPluginKey)`. It is true if you use strings directly, e.g. `tr.getMeta('sync')`, those should unique to not conflict with other plugins.